x-pack/plugins/observability/server/domain/services/date_range.test.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TimeWindow } from '../../types/models/time_window';
+import { Duration, DurationUnit } from '../../types/schema';
+import { toDateRange } from './date_range';
+
+const THIRTY_DAYS = new Duration(30, DurationUnit.d);
+const WEEKLY = new Duration(1, DurationUnit.w);
+const BIWEEKLY = new Duration(2, DurationUnit.w);
+const MONTHLY = new Duration(1, DurationUnit.M);
+const QUARTERLY = new Duration(1, DurationUnit.Q);
+
+const NOW = new Date('2022-08-11T08:31:00.000Z');
+
+describe('toDateRange', () => {
```

Comment:
Shall we also add a test for future starting time?

> throw new Error('Cannot compute date range with future starting time');

Reply:
Good idea!